builtin/logical/pki/ocsp_test.go

```diff
@@ -309,6 +309,63 @@ func TestOcsp_RevokedCertHasIssuerWithoutAKey(t *testing.T) {
 	require.Equal(t, ocsp.UnauthorizedErrorResponse, respDer)
 }
 
+// Verify if for some reason an end-user has rotated an existing certificate using the same
+// key so our algo matches multiple issuers and one has OCSP usage disabled. We expect that
+// even if a prior issuer issued the certificate, the new matching issuer can respond and sign
+// the response to the caller on its behalf.
+//
+// NOTE: This test is a bit at the mercy of iteration order of the issuer ids.
+//       If it becomes flaky, most likely something is wrong in the code
+//       and not the test.
+func TestOcsp_MultipleMatchingIssuersOneWithoutSigningUsage(t *testing.T) {
+	b, s, testEnv := setupOcspEnv(t, "ec")
+
+	// Create a matching issuer as issuer1 with the same backing key
+	resp, err := CBWrite(b, s, "root/rotate/existing", map[string]interface{}{
+		"key_ref":     testEnv.keyId1,
+		"ttl":         "40h",
+		"common_name": "example-ocsp.com",
+	})
+	requireSuccessNonNilResponse(t, resp, err, "rotate issuer failed")
+	requireFieldsSetInResp(t, resp, "issuer_id")
+	rotatedCert := parseCert(t, resp.Data["certificate"].(string))
+
+	// Remove ocsp signing from our issuer
+	resp, err = CBPatch(b, s, "issuer/"+testEnv.issuerId1.String(), map[string]interface{}{
+		"usage": "read-only,issuing-certificates,crl-signing",
+	})
+	requireSuccessNonNilResponse(t, resp, err, "failed resetting usage flags on issuer")
+	requireFieldsSetInResp(t, resp, "usage")
+	// Do not assume a specific ordering for usage...
+	usages, err := NewIssuerUsageFromNames(strings.Split(resp.Data["usage"].(string), ","))
+	require.NoError(t, err, "failed parsing usage return value")
+	require.True(t, usages.HasUsage(IssuanceUsage))
+	require.True(t, usages.HasUsage(CRLSigningUsage))
+	require.False(t, usages.HasUsage(OCSPSigningUsage))
+
+	// Request an OCSP request from it, we should get an Unauthorized response back
```

Comment:
This comment is out of place, and not what this test is checking?

Reply:
Yup sadly copy/paste let me down, going a little too quickly. Thanks for feedback, the other items have been addressed in the feedback commit. Much appreciated!